packages/load-test/helpers.ts

```diff
@@ -0,0 +1,189 @@
+    while (true) {
+      try {
+        const tx = await l2TxStorage
+          .connect(wallet)
+          .sendTx(i, Math.round(Date.now() / 1000))
+        await tx.wait()
+        // console.log('sent tx', i, tx.hash)
+        break
+      } catch (error) {
+        console.error('error sending tx', i, 'retrying in 1 second')
+        await sleep(1000)
+      }
+    }
+  }
+}
+
+export const verifyL2Txs = async (l2TxStorage) => {
+  const numTxs = (await l2TxStorage.numTxs()).toNumber()
+  console.log('found', numTxs, 'txs')
+  for (let i = 0; i < numTxs; i++) {
+    const getTx = await l2TxStorage.l2Txs(i)
+    expect(getTx.txIndex.toNumber()).to.equal(
+      i,
+      'tx sent index does not match received index in storage'
+    )
+    const receivedTime = getTx.timestampReceived.toNumber()
+    const sentTime = getTx.realWorldTimeSent.toNumber()
+    // if (sentTime > receivedTime) {
+    //   console.log(
+    //     `received tx ${i}: ${sentTime - receivedTime} seconds in the past`
+    //   )
+    // } else {
+    //   console.log(`received tx ${i} after `, receivedTime - sentTime, 'seconds')
+    // }
+  }
+}
+
+export const verifyL2Deposits = async (
+  l1DepositInitiator,
+  l2DepositTracker,
+  walletAddress,
+  actualIndexes
+) => {
+  const numInitiatedDeposits = (
+    await l1DepositInitiator.numDeposits()
+  ).toNumber()
+  const numDeposits = (await l2DepositTracker.numDeposits()).toNumber()
+
+  console.log('found', numDeposits, 'completed deposits')
+  for (let i = 0; i < numDeposits; i++) {
+    const deposit = await l2DepositTracker.l2Deposits(i)
+    expect(deposit.depositerAddress).to.equal(
+      walletAddress,
+      'Received Depositer Address does not match up'
+    )
+    const receivedTime = deposit.receivedTimestamp.toNumber()
+    const sentTime = deposit.initiatedTimestamp.toNumber()
+    // console.log(
+    //   `deposit ${i} received after ${receivedTime - sentTime} seconds`
+    // )
+    expect(deposit.depositIndex.toNumber()).to.equal(
+      actualIndexes[i],
+      'Received Deposit index does not match up to array index'
+    )
+  }
+}
+
+export const verifyL1Deposits = async (l1DepositInitiator, walletAddress) => {
+  const numDeposits = (await l1DepositInitiator.numDeposits()).toNumber()
+  console.log('found', numDeposits, ' initiated deposits')
+  const actualIndexes = []
+  for (let i = 0; i < numDeposits; i++) {
+    const deposit = await l1DepositInitiator.l1Deposits(i)
+    expect(deposit.depositerAddress).to.equal(
+      walletAddress,
+      'Sent Depositer Address does not match up'
+    )
+    if (deposit.depositIndex.toNumber() !== i) {
+      console.error(
+        `Sent Deposit index ${deposit.depositIndex.toNumber()} does not match up to array index ${i}`
+      )
+    }
+    actualIndexes.push(deposit.depositIndex.toNumber())
+    // expect(deposit.depositIndex.toNumber()).to.equal(i, 'Sent Deposit index does not match up to array index')
```

Comment:
Why is this assertion commented out?

Reply:
was originally commented out since not awaiting on deposits caused them to sometimes be processed out of order on Kovan. Added awaits now, so should be fine to add back in